Static member using type parameter emits malformed JS and Go

Trying to compile and run the following Dafny program works for C#, but produces crashing code for JavaScript and malformed code for Go.

```
class C<X(0)> {
  static const u: X
}

method Main() {
  print C<int>.u, "\n";
  print C<nat>.u, "\n";
  print C<real>.u, "\n";
  print C<bool>.u, "\n";
}
```

It works for C#:
```
$ dafny /compile:3 test.dfy /compileTarget:cs
Dafny 2.3.0.10506

Dafny program verifier finished with 0 verified, 0 errors
Running...

0
0
0.0
false
```

Repro for JavaScript:

```
$ dafny /compile:3 test.dfy /compileTarget:js
Dafny 2.3.0.10506

Dafny program verifier finished with 0 verified, 0 errors
Running...

[stdin]:1044
      return this.rtd$_X.Default;
                         ^

TypeError: Cannot read property 'Default' of undefined
    at Function.get u [as u] ([stdin]:1044:26)
    at Function.Main ([stdin]:1054:56)
    at [stdin]:1068:19
    at Script.runInThisContext (vm.js:123:20)
    at Object.runInThisContext (vm.js:312:38)
    at Object.<anonymous> ([stdin]-wrapper:6:22)
    at Module._compile (internal/modules/cjs/loader.js:721:30)
    at evalScript (internal/bootstrap/node.js:720:27)
    at Socket.process.stdin.on (internal/bootstrap/node.js:391:9)
    at Socket.emit (events.js:193:15)
```

Repro for Go:
```
$ dafny /compile:3 test.dfy /compileTarget:go
Dafny 2.3.0.10506

Dafny program verifier finished with 0 verified, 0 errors
Compiled program written to test-go/src/test.go
Additional code written to test-go/src/dafny/dafny.go
Additional code written to test-go/src/System_/System_.go
Running...

# command-line-arguments
test-go/src/test.go:28:6: undefined: _this
```

--- a/Test/git-issues/git-issue-354.dfy
+++ b/Test/git-issues/git-issue-354.dfy
@@ -1,0 +1,18 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:cs "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:js "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:go "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:java "%s" >> "%t"
+// RUN: %diff "%s.expect" "%t"
+
+class C<X(0)> {
+  static const u: X
+}
+
+method Main() {
+  print C<int>.u, "\n";
+  print C<nat>.u, "\n";
+  print C<real>.u, "\n";
+  print C<bool>.u, "\n";
+}
+

--- a/Test/git-issues/git-issue-354.dfy.expect
+++ b/Test/git-issues/git-issue-354.dfy.expect
@@ -1,0 +1,26 @@
+
+Dafny program verifier finished with 0 verified, 0 errors
+
+Dafny program verifier finished with 0 verified, 0 errors
+0
+0
+0.0
+false
+
+Dafny program verifier finished with 0 verified, 0 errors
+0
+0
+0.0
+false
+
+Dafny program verifier finished with 0 verified, 0 errors
+0
+0
+0.0
+false
+
+Dafny program verifier finished with 0 verified, 0 errors
+0
+0
+0.0
+false

